rfd/00xx-pod-access-requests.md

```diff
@@ -0,0 +1,598 @@
+Teleport will modify his Kubernetes RBAC principals to include a new group that grants
+him access to the Pod. This indeed will extend his permissions when interacting
+with the Kubernetes Cluster instead of restricting them as in approach 1.
+
+To do this, Teleport will create temporary `Roles` in the Kubernetes cluster and
+will associate them with Kubernetes groups that the user can impersonate.
+
+Once Teleport receives an access request to a Pod, it will create the Kubernetes
+RBAC `Role` and will create `RoleBinding` to assign it to an RBAC group. Afterward,
+Teleport will include the newly created group in the user's certificate.
+
+Once the user has the certificate with the new group embedded, Teleport will
+forward the certificate details into the Kubernetes cluster as Impersonation Headers.
+
+When the access request expires, Teleport will clean the temporary Kubernetes
+RBAC `Role` and `RoleBinding` objects.
+
+The final decision was to implement the first option, however, the second option
+details are kept in this RFD.
+
+## UX: Request Flow
+
+Extending [RFD 59](https://github.com/gravitational/teleport/blob/master/rfd/0059-search-based-access-requests.md),
+Teleport will consider Kubernetes Pods as native resources.
+This allows users to search Pods that normally they do not
+have access to and request access to them. Besides requesting access to
+specific Pods - using the Pod full name, Teleport allows users to request
+access to Pods that follow a pattern like Pods automatically created from a
+Deployment - `deploymentname-*-*`.
+
+```bash
+$ tsh request search --kind pod --search pod
+
+Found 3 items:
+name kind     id
+pod-1 pod     pod:pod-1
+dep-1 pod     node:dep-1
+dep-2 pod     node:dep-2
+
+$ tsh request create --resources "pod:pod1,pod:dep-*"
+Waiting for request to be approved...
+Approved!
+$ kubectl get pods 
+```
+
+Similarly to Resource Based Access Requests, Pod Access requests will leverage
+`search_as_roles` to control which Pods the user can view while searching.
+
+```yaml
+kind: role
+metadata:
+  name: response-team
+spec:
+  allow:
+    request:
+      # search_as_roles allows a member of the response team
+      # to search for resources accessible to users with the kube-admins role,
+      # which they will be allowed to request access to. This includes Pods in 
+      # the target Kubernetes Cluster.
+      search_as_roles: [kube-admin]
+```
+
+```yaml
+kind: role
+metadata:
+  name: kube-admin
+spec:
+  allow:
+    # kubernetes_groups will be used by Teleport to fetch the Pods available in 
+    # the target cluster. The kubernetes_groups field is used by Kubernetes RBAC 
+    # to limit the Pods the user can request access to.
+    kubernetes_groups: ["system:masters"]
+    # kubernetes_labels defines which Kubernetes clusters the user is able to 
+    # request Pod access into.
+    kubernetes_labels:
+       owner: prod_team
```

Comment:
Good catch. It's a typo.
For roles <=V5, the field is automatically populated with "*" but for a V6 role you must define the value otherwise the request won't return anything since everything will be filtered out.

I need to update the RFD to include the namespace filtering as well - "namespace/*" or "*/somepod-*".